elastic/eui
# feat: add JSON docgen generator to feed docusaurus website with accurate prop definitions

## Summary

Resolves https://github.com/elastic/eui/issues/7926

This PR adds a new private package `@elastic/eui-docgen` that generates docgen JSON files for all EUI source files just like the [babel script](https://github.com/elastic/eui/blob/main/packages/eui/scripts/babel/react-docgen-typescript.js) did. It reuses most of the old code and results in lots of definition JSON files being generated. This PR also adds an example page to docusaurus and very, very basic `PropTable` component to the `docusaurus-theme` package. These will be updated and improved in following PRs.

## QA

1. Checkout this PR and run `yarn`
2. Go to `packages/website` and build all of its local dependencies: `yarn workspaces foreach -Rpt --from @elastic/eui-website run build`
3. Run the development server `yarn start`
4. Confirm the test page renders a basic prop table
